src/Transports.AspNetCore/GraphQLHttpMiddlewareOptions.cs

```diff
@@ -0,0 +1,107 @@
+#nullable enable
+
+using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Configuration options for <see cref="GraphQLHttpMiddleware"/>.
+/// </summary>
+public class GraphQLHttpMiddlewareOptions
+{
+    /// <summary>
+    /// Enables handling of GET requests.
+    /// </summary>
+    public bool HandleGet { get; set; } = true;
+
+    /// <summary>
+    /// Enables handling of POST requests, including form submissions, JSON-formatted requests and raw query requests.
+    /// </summary>
+    public bool HandlePost { get; set; } = true;
+
+    /********** WebSockets support ********
+    
+    /// <summary>
+    /// Enables handling of WebSockets requests.
+    /// <br/><br/>
+    /// Requires calling <see cref="WebSocketMiddlewareExtensions.UseWebSockets(IApplicationBuilder)"/>
+    /// to initialize the WebSocket pipeline within the ASP.NET Core framework.
+    /// </summary>
+    public bool HandleWebSockets { get; set; } = true;
+
+    ******************/
+
+    /// <summary>
+    /// Enables handling of batched GraphQL requests for POST requests when formatted as JSON.
+    /// </summary>
+    public bool EnableBatchedRequests { get; set; } = true;
+
+    /// <summary>
+    /// Enables parallel execution of batched GraphQL requests.
+    /// </summary>
+    public bool BatchedRequestsExecuteInParallel { get; set; } = true;
```

Comment:
I'm fine to just allow parallel execution in that PR without fine-grained tuning around concurrent limit. I posted an issue about that to do not forget.